pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -569,14 +578,41 @@ def getData(self):
                         x = x[x0:x1]
                         y = y[x0:x1]
                     
-            if ds > 1:
+        
+            if len(x) > 1 and ds > 1:
+                n = len(x) // ds
+                if self.opts['quantizedDownsample']:
+                    i0 = np.searchsorted( x, ds*(1 + x[0]//ds ))
```

Comment:
np.searchsorted isn't a search for an exact match -- its just a comparison based sort for inserting in-order